lib/AST/ASTImporter.cpp

```diff
@@ -1104,16 +1104,23 @@ bool ASTNodeImporter::ImportDeclParts(NamedDecl *D, DeclContext *&DC,
   FunctionDecl *FunDecl;
   if (isa<RecordDecl>(D) && (FunDecl = dyn_cast<FunctionDecl>(OrigDC)) &&
       FunDecl->hasBody()) {
-    SourceRange RecR = D->getSourceRange();
-    SourceRange BodyR = FunDecl->getBody()->getSourceRange();
+    SourceManager &SM = D->getASTContext().getSourceManager();
```

Comment:
I was thinking about an alternative solution which does not involve source locations:
```
  if (isa<RecordDecl>(D) && (FunDecl = dyn_cast<FunctionDecl>(OrigDC)) &&
      FunDecl->hasBody()) {
    for (const ParmVarDecl *P : FunDecl->parameters()) {
      auto getLeafPointeeType = [](const Type *T) {
        while (T->isPointerType() || T->isArrayType()) {
          T = T->getPointeeOrArrayElementType();
        }
        return T;
      };
      const Type *LeafT =
          getLeafPointeeType(P->getType().getCanonicalType().getTypePtr());
      if (auto * RT = dyn_cast<RecordType>(LeafT)) {
        if (RT->getDecl() == D) {
          Importer.FromDiag(D->getLocation(), diag::err_unsupported_ast_node)
              << D->getDeclKindName();
          Importer.setEncounteredUnsupportedNode(true);
          return true;
        }
      }
    }
  }
```